var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -11,13 +12,15 @@ class Netcdf(Package):
     version('4.4.0', 'f01cb26a0126dd9a6224e76472d25f6c')
     version('4.3.3', '5fbd0e108a54bd82cb5702a73f56d2ae')
 
+    variant('mpi', default=True, description='Enables MPI parallelism')
     variant('fortran', default=False, description="Download and install NetCDF-Fortran")
     variant('hdf4',    default=False, description="Enable HDF4 support")
 
     # Dependencies:
     depends_on("curl")  # required for DAP support
     depends_on("hdf", when='+hdf4')
-    depends_on("hdf5")  # required for NetCDF-4 support
+    depends_on("hdf5+mpi~cxx", when='+mpi')  # required for NetCDF-4 support
```

Comment:
i asked myself exactly the same question here https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/dealii/package.py#L42 . I agree with @eschnett that `~cxx` should either be commented or removed.